lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -1301,15 +1346,20 @@ void PatchEntryPointMutate::determineUnspilledUserDataArgs(ArrayRef<UserDataArg>
                                                            IRBuilder<> &builder,
                                                            SmallVectorImpl<UserDataArg> &unspilledArgs) {
 
-  SmallVector<UserDataArg, 1> spillTableArg;
+  Optional<UserDataArg> spillTableArg;
+
+  // In compute-with-calls, we need to ensure that the compute shader and library code agree that s15 is the spill
+  // table pointer, even if it is not needed, because library code does not know whether a spill table pointer is
+  // needed in the pipeline. Thus we cannot use s15 for anything else. Using the single-arg UserDataArg
+  // constructor like this means that the arg is not used, so it will not be set up in PAL metadata.
 
   auto userDataUsage = getUserDataUsage(m_shaderStage);
   if (!userDataUsage->spillTable.users.empty() || userDataUsage->pushConstSpill ||
-      userDataUsage->spillUsage != UINT_MAX) {
+      userDataUsage->spillUsage != UINT_MAX || isComputeWithCalls()) {
```

Comment:
Why do we need this change? Is it because, in the event of a compute library that decides lower down that we need spilling, spillTableArg was never getting set up properly with UserDataMapping::SpillTable and spillTable.entryArgIdx? I think it would be better to fix that problem, without introducing an unconditional spill table in pal metadata here.

Reply:
Right, I missed that.